File: fluentcore/src/main/java/com/mcreater/fxfluent/util/interpolatables/DoubleNumberInterpolatable.java
package com.mcreater.fxfluent.util.interpolatables;

import javafx.animation.Interpolatable;
import javafx.util.Pair;

public class DoubleNumberInterpolatable extends Pair<Double, Double> implements Interpolatable<DoubleNumberInterpolatable> {
    public DoubleNumberInterpolatable(Double aDouble, Double aDouble2) {
        super(aDouble, aDouble2);
    }

    public DoubleNumberInterpolatable interpolate(DoubleNumberInterpolatable d, double v) {
        return new DoubleNumberInterpolatable(
                getKey() - (d.getKey() - getKey()) * v,
                getValue() - (d.getValue() - getValue()) * v
        );
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/interpolatables/DoubleNumberInterpolatable.kt
package com.mcreater.fxfluent.util.interpolatables

import javafx.animation.Interpolatable
import javafx.util.Pair

class DoubleNumberInterpolatable(k: Double, v: Double) : Pair<Double, Double>(k, v), Interpolatable<DoubleNumberInterpolatable> {
    override fun interpolate(k: DoubleNumberInterpolatable, v: Double): DoubleNumberInterpolatable {
        return DoubleNumberInterpolatable(
            key - (k.key - key) * v,
            value - (k.value - value) * v
        )
    }
}